app/controllers/challenge_signups_controller.rb

```diff
@@ -127,25 +127,32 @@ def index
   end
 
   def summary
+    @summary = ChallengeSignupSummary.new(@collection)
+
     if @collection.signups.count < (ArchiveConfig.ANONYMOUS_THRESHOLD_COUNT/2)
       flash.now[:notice] = ts("Summary does not appear until at least %{count} sign-ups have been made!", count: ((ArchiveConfig.ANONYMOUS_THRESHOLD_COUNT/2)))
     elsif @collection.signups.count > ArchiveConfig.MAX_SIGNUPS_FOR_LIVE_SUMMARY
       # too many signups in this collection to show the summary page "live"
-      if !File.exists?(ChallengeSignup.summary_file(@collection)) ||
-          (@collection.challenge.signup_open? && File.mtime(ChallengeSignup.summary_file(@collection)) < 1.hour.ago)
-        # either the file is missing, or signup is open and the last regeneration was more than an hour ago.
+      modification_time = @summary.cached_time
+
+      # The time is always written alongside the cache, so if the time is
+      # missing, then the cache must be missing as well -- and we want to
+      # generate it. We also want to generate it if signups are open and it was
+      # last generated more than an hour ago.
+      if modification_time.nil? ||
+         (@collection.challenge.signup_open? && modification_time < 1.hour.ago)
 
-        # touch the file so we don't generate a second request
-        summary_dir = ChallengeSignup.summary_dir
-        FileUtils.mkdir_p(summary_dir) unless File.directory?(summary_dir)
-        FileUtils.touch(ChallengeSignup.summary_file(@collection))
+        # Touch the cache so that we don't try to generate the summary a second
+        # time on subsequent page loads.
+        @summary.touch_cache
```

Comment:
I can't see how this works when the code can be run on lost of different servers ( I can see there is no migrations so the shared state is not in mysql ).

I can imagine writing a please wait message in to memcached which would achive this ?

Reply:
Ariana pointed out its a methed call.... sorry